### ISSUE TITLE
set hostname -f in nodes

### ISSUE THREAD

**Issue by ajohnstone** [kubernetes/kops#792]
When launching a daemonset with ```hostNetwork: true```, ```hostname -f``` fails within pods and on the ec2 instance itself.

E.g.

```
kubectl --context=production-a-cluster.k8s-production.photobox.com --namespace=kube-system exec --tty -i sensu-client-8vk3t bash
root@ip-10-119-100-62:/# hostname                                                                                                                                                                                                     
ip-10-119-100-62

root@ip-10-119-100-62:/# hostname -f
hostname: Name or service not known
```

If you ssh into the nodes and append entries to /etc/hosts hostname -f will succeed and additionally inside a container on that host with ```hostNetwork: true```.

As a result its suggested to bootstrap a EC2 node with the appended hostnames to /etc/hosts
This script should work... https://gist.github.com/ajohnstone/616b3f316b8d60af56abe9efb1f70feb

Just to note enableDnsHostnames & enableDnsSupport is set to true on the VPC.

Using kubenet

cc @justinsb

**Comment by chrislovecnm** [comment]
@ajohnstone is this required for CNI?

**Comment by ajohnstone** [comment]
/etc/cloud/templates/hosts.debian.tmpl - could also set manage_etc_hosts: True in cloud-init.

"use host networking " - I meant hostNetwork, half asleep :)

**Comment by chrislovecnm** [comment]
@ajohnstone ping me when you are rested ... still not understanding.

**Comment by ajohnstone** [comment]
Another example of this biting 
```
helm list
Error: forwarding ports: error upgrading connection: error dialing backend: dial tcp: lookup ip-10-117-92-2.ec2.internal on 10.117.0.2:53: no such host
```

**Comment by ajohnstone** [comment]
Fixed by correcting DHCP option set #1556
